### PR TITLE
tools/insufficient_memory: restore build with no docker

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

It's useful being able to run this script without having Docker installed.
Introduce the `--no-docker` parameter to run the script without docker.

### Testing procedure

`add_insufficient_memory_board.sh --no-docker` compiles all apps for a board without the use of Docker.

### CODE REVIEW

File: dist/tools/insufficient_memory/add_insufficient_memory_board.sh [hunk 1]
```diff
@@ -30,21 +32,25 @@ else
     CRESET=
 fi
 
+if [ "$1" == "--no-docker" ]; then
+    shift 1
+else
+    # Use a standardized build within Docker and with minimal output
+    export DOCKER_MAKE_ARGS=${MAKE_ARGS}
+    export RIOT_CI_BUILD=1
```
Comment (aabadie) [inline dist/tools/insufficient_memory/add_insufficient_memory_board.sh:40]: `RIOT_CI_BUILD` could also apply without Docker I think.

File: dist/tools/insufficient_memory/add_insufficient_memory_board.sh [hunk 2]
```diff
@@ -30,21 +32,25 @@ else
...
-# Use a standardized build within Docker and with minimal output
-export BUILD_IN_DOCKER=1
-export DOCKER_MAKE_ARGS="-j4"
-export RIOT_CI_BUILD=1
-
 for app in ${APPLICATIONS}; do
     application=$(dirname ${app})
     printf "${CNORMAL}%-40s${CRESET}" ${application}
-    output=$(make BOARD=${BOARD} -C ${RIOTBASE}/${application} 2>&1)
+    output=$(make BOARD=${BOARD} ${MAKE_ARGS} -C ${RIOTBASE}/${application} 2>&1)
```
Comment (aabadie) [inline dist/tools/insufficient_memory/add_insufficient_memory_board.sh:53]: When using docker, `MAKE_ARGS` is added twice in the command line (here and via `DOCKER_MAKE_ARGS`). You could have a third variable, used in the command line that is set to `MAKE_ARGS` without docker, and empty in docker.

Comment (benpicco) [reply dist/tools/insufficient_memory/add_insufficient_memory_board.sh:55]: I added a `jobs` variable

**Review by aabadie — APPROVED** [review]
Code changes are good and are working as expected (with and without docker).

ACK

**Pull request merged**